scripts/lib/aws_apigwv2_testutil.sh

```diff
@@ -0,0 +1,189 @@
+#!/usr/bin/env bash
+
+###########################################
+# API
+###########################################
+create_http_api_and_validate() {
+## create api resource
+cat <<EOF | kubectl apply -f - >/dev/null 2>&1
+apiVersion: apigatewayv2.services.k8s.aws/v1alpha1
+kind: API
+metadata:
+  name: $api_name
```

Comment:
I *strongly* recommend not using environment variables (and assuming that a particular env var is set to some non-nil value). Instead, I recommend using function local-scoped variables and actually pass arguments to the Bash function. Then, make a docstring comment that outlines expected parameters.

Example:

https://github.com/aws/aws-controllers-k8s/blob/main/scripts/lib/testutil.sh#L5-L23

Reply:
I had it that way and then I added `set -u` on the e2e test and it made things way more readable. 

If you still think local variable would be the way to go instead of `set -u`, I can make that change.

Reply:
It's cool to clean it up in a followup PR.